src/main/java/bootstrap/EntityComponentScanner.java

```diff
@@ -0,0 +1,68 @@
+package bootstrap;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EntityComponentScanner {
+    private static final String DOT = ".";
+    private static final String CLASS_FILE_EXTENSION = DOT + "class";
+
+    public List<Class<?>> scan(String basePackage) throws ClassNotFoundException {
+        List<Class<?>> classes = new ArrayList<>();
```

Comment:
함수내에서 상태를 선언하는 순간, 메서드가 점점 여러 책임을 가지게 되며 생각이 복잡해지는것 같습니다.

`List<Class<?>> classes = new ArrayList<>();` 를 없애보고 다시 바라본다면 scan함수가 좀 더 책임이 분명해지고 가벼워지지 않을까 생각이되어지네요 ㅎ

`scan 메서드`를 가능한 작게 유지하여 단일 책임 원칙을 준수하도록 개선해 보면 좋을것 같습니다 🥹 

```suggestion
    public List<Class<?>> scan(String basePackage) throws ClassNotFoundException {
        File baseDir = resolveBaseDirectory(basePackage);

        if (!isValidDirectory(baseDir)) {
            return new ArrayList<>();
        }
        
        ...
```